test/binary/binary_test.go

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright © 2022 Ondřej Míchal
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package binary
+
+import (
+	"debug/elf"
```

Comment:
How does Go's `debug/elf` compare with other implementations like GNU Binutils, in terms of architecture support and the ability to handle a diverse variety of ELFs?

Reply:
Overall I'd say the support is lower than of `readelf` from GNU Binutils. From what I understood though, `debug/elf` supports both [32-bit and 64-bit architectures](https://pkg.go.dev/debug/elf#Class) and recognizes many machine types and many OS ABIs. Though, am not sure how the last two parts translate into "architecture support". Most likely didn't answer your question in full.

Probably is noteworthy that the tool is for checking the Toolbx binary and not an arbitrary binary. My presumption is that whether Go is capable of producing a binary on a certain architecture, the created binary also has to be processable by the `debug/elf` package.